src/libraries/System.Text.Json/gen/JsonSourceGenerator.cs

```diff
@@ -67,7 +67,7 @@ private sealed class SyntaxReceiver : ISyntaxReceiver
 
             public void OnVisitSyntaxNode(SyntaxNode syntaxNode)
             {
-                if (syntaxNode is ClassDeclarationSyntax cds)
+                if (syntaxNode is ClassDeclarationSyntax { AttributeLists.Count: > 0, BaseList.Types.Count: > 0 } cds)
                 {
                     (ClassDeclarationSyntaxList ??= new List<ClassDeclarationSyntax>()).Add(cds);
```

Comment:
(nit) I wonder if this property should be renamed. Before it held all classes, so the name made more sense. But now it only holds "potential context classes", so maybe a rename is in order?